### ISSUE TITLE
Add imgpath property to all builders

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1571]
Currently, image path differs according to builder.
To get recommended image path, we have to write same code in each extensions.
For example, `sphinx.ext.graphviz` has code like below::

```
    if hasattr(self.builder, 'imgpath'):
        # HTML
        relfn = posixpath.join(self.builder.imgpath, fname)
        outfn = path.join(self.builder.outdir, '_images', fname)
    else:
        # LaTeX
        relfn = fname
        outfn = path.join(self.builder.outdir, fname)
```

And, I wrote copied codes to many extensions which generate image files.

I think this code is dirty and waste.
It would be better to add `imgpath` property to all builders.
It will make image-extensions simple.

What do you think about it ?

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1571
- Originally reported by: Takeshi Komiya
- Originally created at: 2014-09-18T04:58:00.981

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2014-09-18 03:10:32+00:00_

I think the idea is very good. We will proceed to include it to Sphinx-1.3.

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2014-09-28 15:38:48+00:00_

https://bitbucket.org/birkenfeld/sphinx/pull-request/297 has been merged. Update CHANGES for it. Closes #1571.

→ <<cset 86ff9589bcb11a2779f4faf0f95eeb6af430882c>>
